File: Go-Solutions/nowcoder/jz15.go
package nowcoder

func ReverseList(pHead *ListNode) *ListNode {
	// write code here
	var newHead *ListNode = nil
	var next *ListNode = nil
	runner := pHead
	for runner != nil {
		next = runner.Next
		runner.Next = newHead
		newHead = runner
		runner = next
	}
	return newHead
}

File: Java-Kotlin/src/main/java/jianzhi/jz15.java
package jianzhi;

import common.datastructure.ListNode;

public class jz15 {

    public ListNode ReverseList(ListNode head) {
        ListNode newHead = null;
        ListNode next = null;
        ListNode cur = head;
        while (cur != null) {
            next = cur.next;
            cur.next = newHead;
            newHead = cur;
            cur = next;
        }
        return newHead;
    }
}
